### PR TITLE
Replace "Return" with "Next" on keyboard to move between fields

### LINKED ISSUE #87: Replace "Return" with "Next" on keyboard to move between fields
As per https://docs.google.com/document/d/1K-iSxSOaCVdMJweiwOUwVO92tVaCU7FjvlUmco11tgY/edit#bookmark=id.a3z6w9dp6yu

"Return" implies a line break. 

Can we have the keyboard say “next” on the button right, and take users to the next field? 

If not, have it say “done” instead of “return,” because “return” may imply a line break.

This change should be "global" for all one-line text fields (but not for text areas like notes where a line break might be wanted).

Steps to reproduce:
Board vessel
click "add new vessel"
Tap "Vessel Name"
Keyboard shows up with "return" in the lower right. If you click it, it will act as 'Done'.

### PR DESCRIPTION
**Pull request by pankova**
## Related Issue

Fixes #87 

## Checklist:

- [x] I have read the [contributor's guide](https://wildaid.github.io/contribute/index.html).
- [x] I linked an issue in the previous section
- [x] I have commented on the linked issue
- [x] I was assigned the linked issue (not required)
- [x] I have tested the change to the best of my ability against the [sandbox](https://wildaid.github.io/contribute/sandbox.html) or a [local build](https://wildaid.github.io/build).

Optional items:

- [ ] My change adds new text and requires a change to translations.
- [ ] My change requires a change to the documentation.
- [ ] I have submitted a PR to the [documentation repo](https://github.com/WildAid/wildaid.github.io).
- [ ] I was not able to test... (explain below, e.g. you did not have permissions to test a specific feature)
- [ ] This change depends O-FISH Realm repository changes (explain below)
- [ ] This change depends O-FISH Web repository changes (explain below)

* **Optional: Add any explanations here** 
Now in SwiftUI, there are no opportunities for TextField: 

1. to set `returnKeyType`
2. to move to next text filed on the screen after return key tap

I couldn't fix it by SwiftUI methods, so the main way (as approved by the internet community) is to create `UITextField -> SwiftUI` adapter. It is realized in this pull request :) The main properties as text color, keyboard type, autocapitalization type, etc are taken into account.

* **Optional: Add any relevant screenshots here**

### CODE REVIEW

File: o-fish-ios/Views/Components/InputField.swift [hunk 1]
```diff
@@ -11,15 +11,18 @@ struct InputField: View {
...
+
     var showingSecureField = false
     var showingWarning = false
 
     var captionColor = Color.removeAction
     var separatorColor = Color.inactiveBar
     var warningColor = Color.spanishOrange
 
-    var inputChanged: ((String) -> Void)?
+    var inputChanged: (() -> Void)?
```
Comment (pankova) [inline o-fish-ios/Views/Components/InputField.swift:23]: `String` value wasn't used anywhere, so I removed it

**Comment by Sheeri** [comment]
@pankova this is failing with:
/Users/runner/work/o-fish-ios/o-fish-ios/o-fish-ios/Views/Components/TextField/FocusableTextFieldAdapter.swift:49:31: argument labels '(_:)' do not match any available overloads

Can you make it so that's not an issue?

**Comment by pankova** [comment]
> @pankova this is failing with:
> /Users/runner/work/o-fish-ios/o-fish-ios/o-fish-ios/Views/Components/TextField/FocusableTextFieldAdapter.swift:49:31: argument labels '(_:)' do not match any available overloads
> 
> Can you make it so that's not an issue?

The deployment target is **iOS 14.0**, isn't it? 
This value is from [project file](https://github.com/WildAid/o-fish-ios/blob/main/o-fish-ios.xcodeproj/project.pbxproj):
```
IPHONEOS_DEPLOYMENT_TARGET = 14.0; 
```

But in Fastlane log I see:
```
The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 14.0, but the range of supported deployment target versions is 8.0 to 13.7.99. (in target 'o-fish-ios' from project 'o-fish-ios')
```
-is it OK? 


Log error:
```
[18:33:45]: ▸ ❌  /Users/runner/work/o-fish-ios/o-fish-ios/o-fish-ios/Views/Components/TextField/FocusableTextFieldAdapter.swift:49:31: argument labels '(_:)' do not match any available overloads
531
[18:33:45]: ▸         textField.textColor = UIColor(Color.text)
532
[18:33:45]: ▸                               ^      ~~~~~~~~~~~~
```
is caused by unrecognized UIColor [init](https://developer.apple.com/documentation/uikit/uicolor/3550899-init), which is available from iOS 14.0. 

Maybe we could fix the minimum iOS version in some fastlane config?

**Comment by o-fish-wildaid** [comment]
[Review PRs](https://trello.com/c/R2aGCG3t/451-review-prs)

**Comment by pankova** [comment]
>  [Review PRs](https://trello.com/c/R2aGCG3t/451-review-prs)

Just in case, I have no right to see it

**Comment by pankova** [comment]
@Sheeri hello 🙂 Are there some news with fastlane? Could I help in some way?

**Comment by pankova** [comment]
@Sheeri @am-MongoDB sorry for bothering, but this is my last Hacktoberfest PR. Could you please answer about fastlane and review It? 🙏

**Comment by pankova** [comment]
UPD: I have rebased onto the main and all became OK. Sorry for panic in an empty place with fastane 🙈

**Comment by Sheeri** [comment]
Hi @pankova - I have set this with a label of "hacktoberfest-accepted" so it should show up on your hacktoberfest dashboard at https://hacktoberfest.digitalocean.com/profile - please let me know if it's not showing up.

It looks like the fastlane check has been fixed. 

We are discussing whether or not to move forward with merging this code - the code itself is great; however, since there is no real way to do this in SwiftUI, we are weighing whether or not we want to introduce this code (it's a lot of code for such a small feature).

This in no way reflects on your contribution, which is why we have set the label as accepted. We had another feature that is related, that was put into the code, that dismisses the keyboard when the user scrolls down - we are thinking that may be enough for the user and we don't want to introduce so much code just to have a 'next' button.

We really appreciate your contribution! Know that even if we don't merge the code, your PR made us really think about what is best for the user, and what tradeoffs we want to make.

**Comment by pankova** [comment]
@Sheeri thanks for the answer! I'm really was glad to help. Unfortunately, SwiftUI has no UIResponder features. I think next time I should warn about an indirect decision. Thanks for your warm atmosphere and great project mission! 💪

**Review by am-MongoDB — CHANGES_REQUESTED** [review]
Works great – just spotted one glitch that wasn't there before – if you get the username/password wrong then the password field still shows as being filled in (while the login button is deactivated). When the username/password is wrong, the view should show the password field as being empty.

image

**Comment by pankova** [comment]
@am-MongoDB I'll try to fix it today! Thanks!

**Comment by pankova** [comment]
@am-MongoDB It's UIKIt+SwiftUI issue again( Sorry, I haven't fix it for 2 hours and could continue only from Monday

**Comment by am-MongoDB** [comment]
@pankova thanks for looking into it. An acceptable workaround, would be to not clear the password when a login fails - i.e. remove `self.password` from this snippet in `LoginView.swift`...

```
                Alert(title: Text("Login Error"),
                    message: Text(errorMessage),
                    dismissButton: .default(Text("Ok")) {
                        self.errorMessage = ""
                        self.password = ""
                    })
```

**Comment by pankova** [comment]
@am-MongoDB I'll try to fix it, but keep this option in mind. Thanks!

**Comment by pankova** [comment]
@am-MongoDB hello! Everything seems to be fine now 🙂

**Review by am-MongoDB — APPROVED** [review]

**Comment by o-fish-wildaid** [comment]
[Review O-FISH PR](https://trello.com/c/C9dLdYs2/471-review-o-fish-pr)

**Pull request merged**